pupil_src/shared_modules/player_methods.py

```diff
@@ -194,23 +195,124 @@ def correlate_data(data, timestamps):
 
 
 def load_meta_info(rec_dir):
-    meta_info_path = os.path.join(rec_dir, "info.csv")
-    with open(meta_info_path, "r", encoding="utf-8") as csvfile:
-        meta_info = csv_utils.read_key_value_file(csvfile)
-    return meta_info
+    return Pupil_Recording(rec_dir).meta_info
 
 
 def is_pupil_rec_dir(rec_dir):
-    if not os.path.isdir(rec_dir):
-        logger.error("No valid dir supplied ({})".format(rec_dir))
-        return False
     try:
-        meta_info = load_meta_info(rec_dir)
-        meta_info["Recording Name"]  # Test key existence
-    except:
-        logger.error("Could not read info.csv file: Not a valid Pupil recording.")
+        Pupil_Recording(rec_dir)
+        return True
+    except InvalidRecordingException as e:
+        logger.error(str(e))
         return False
-    return True
+
+
+class InvalidRecordingException(Exception):
+    def __init__(self, reason: str, recovery: str = ""):
+        message = reason + "\n" + recovery if recovery else reason
```

Comment:
This prints reason twice in some cases. Is that intended?

Reply:
it's not intended. could you please tell me in which cases you're seeing `reason` twice?

Reply:
I think I misread the priority of the operators. Could you please add round brackets for more explicit grouping?

Reply:
done